action.yml

```diff
@@ -14,36 +14,46 @@ inputs:
 outputs:
   REF_EXISTS:
     description: 'Flag indicating whether the provided reference exists.  true|false'
-    value: '${{ env.REF_EXISTS }}'
+    value: ${{ steps.determine-ref.outputs.REF_EXISTS }}
+  REF_TYPE:
+    description: 'Indicating what type of privided reference: tag|branch|sha'
+    value: ${{ steps.determine-ref.outputs.REF_TYPE }}
 
 runs:
   using: 'composite'
   steps:
-    - shell: bash
+    - id: determine-ref
+      shell: bash
       run: |
         echo "checking ref ${{ inputs.branch-tag-sha }}..."
         ref="${{ inputs.branch-tag-sha }}"
         refWithPrefix="refs/heads${{ inputs.branch-tag-sha }}"
+        
+        echo "::set-output name=REF_EXISTS::false"
+        echo "::set-output name=REF_TYPE::unknown"
 
         if [ $(git tag -l "$ref") ]; then
           echo "The ref '${ref}' exists as a tag"
-          echo "REF_EXISTS=true" >> $GITHUB_ENV
+          echo "::set-output name=REF_EXISTS::true"
+          echo "::set-output name=REF_TYPE::tag"
```

Comment:
Why switch it from an env to an output?  My preference would be to keep it as an env because it is a lot easier to look at.

Reply:
Because it doesn' work.  Or should I say, it doesn't output the env. var.

Reply:
Looks at existing workflows using this action and the REF_EXISTS is empty.